Static build with prometheus remote write backend support

Hi, 

I need to create a static static build with prometheus remote write backend support for an old  debian (squeeze) system.  I'm currently building netdata on debian buster (which has snappy and protobuf libraries installed). The non static build is compiling fine on this system.

My question: **How can I compile netdata as a static binary with prometheus remote write backend support?**

I'm new to netdata and really impressed by it!
Thanks to everyone for the precious work!

Ruben


What I've tried so far:

```
~/netdata# make LDFLAGS="-L/usr/lib/x86_64-linux-gnu/ -static"
~/netdata# ./configure --prefix=/opt/netdata
```

but this fails with:

```
  CC       libnetdata/url/url.o
  CC       libnetdata/json/json.o
  CC       libnetdata/json/jsmn.o
  CC       libnetdata/health/health.o
  CCLD     apps.plugin
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libcrypto.a(b_addr.o): in function `BIO_lookup_ex':
(.text+0xbf3): warning: Using 'getaddrinfo' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libcrypto.a(b_sock.o): in function `BIO_gethostbyname':
(.text+0x71): warning: Using 'gethostbyname' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /usr/bin/ld: DWARF error: could not find abbrev number 84
/tmp/ccD0LSYz.ltrans0.ltrans.o: in function `main':
<artificial>:(.text.startup+0x24ef): warning: Using 'getpwuid' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: <artificial>:(.text.startup+0x269f): warning: Using 'getgrgid' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libcrypto.a(dso_dlfcn.o): in function `dlfcn_globallookup':
(.text+0x11): undefined reference to `dlopen'
/usr/bin/ld: (.text+0x24): undefined reference to `dlsym'
/usr/bin/ld: (.text+0x2f): undefined reference to `dlclose'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libcrypto.a(dso_dlfcn.o): in function `dlfcn_bind_func':
(.text+0x1a8): undefined reference to `dlsym'
/usr/bin/ld: (.text+0x272): undefined reference to `dlerror'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libcrypto.a(dso_dlfcn.o): in function `dlfcn_load':
(.text+0x2e1): undefined reference to `dlopen'
/usr/bin/ld: (.text+0x359): undefined reference to `dlclose'
/usr/bin/ld: (.text+0x395): undefined reference to `dlerror'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libcrypto.a(dso_dlfcn.o): in function `dlfcn_pathbyaddr':
(.text+0x451): undefined reference to `dladdr'
/usr/bin/ld: (.text+0x4b6): undefined reference to `dlerror'
/usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/8/../../../x86_64-linux-gnu/libcrypto.a(dso_dlfcn.o): in function `dlfcn_unload':
(.text+0x682): undefined reference to `dlclose'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:2161: apps.plugin] Error 1
make[2]: Leaving directory '/root/netdata'
make[1]: *** [Makefile:3231: all-recursive] Error 1
make[1]: Leaving directory '/root/netdata'
make: *** [Makefile:1836: all] Error 2
```

--- a/packaging/makeself/install-alpine-packages.sh
+++ b/packaging/makeself/install-alpine-packages.sh
@@ -23,6 +23,7 @@ apk add --no-cache \
     automake \
     gcc \
     make \
+    cmake \
     libtool \
     pkgconfig \
     util-linux-dev \
@@ -34,7 +35,21 @@ apk add --no-cache \
     libuv-dev \
     lz4-dev \
     openssl-dev \
+    snappy-dev \
+    protobuf-dev \
     || exit 1
+
+# snappy doesnt have static version in alpine, let's compile it
+export SNAPPY_VER="1.1.7"
+wget -O /snappy.tar.gz https://github.com/google/snappy/archive/${SNAPPY_VER}.tar.gz
+cd /
+tar -xf snappy.tar.gz
+rm snappy.tar.gz
+cd /snappy-${SNAPPY_VER}
+mkdir build
+cd build
+cmake -DCMAKE_BUILD_SHARED_LIBS=true -DCMAKE_INSTALL_PREFIX:PATH=/usr -DCMAKE_INSTALL_LIBDIR=lib ../
+make && make install
 
 # Judy doesnt seem to be available on the repositories, download manually and install it
 export JUDY_VER="1.0.5"

